lib/rubocop/cop/rspec/leading_subject.rb

```diff
@@ -57,21 +57,16 @@ def check_previous_nodes(node)
 
         private
 
-        def autocorrect(corrector, node)
-          first_node = find_first_offending_node(node)
+        def autocorrect(corrector, node, sibling)
           RuboCop::RSpec::Corrector::MoveNode.new(
             node, corrector, processed_source
-          ).move_before(first_node)
+          ).move_before(sibling)
         end
 
         def offending?(node)
           let?(node) || hook?(node) || example?(node)
         end
 
-        def find_first_offending_node(node)
-          node.parent.children.find { |sibling| offending?(sibling) }
-        end
-
         def in_spec_block?(node)
           node.each_ancestor(:block).any? do |ancestor|
```

Comment:
Observation:
`any?` it feels weird. `first` should work just fine. Can't imagine `subject(...) { }` being defined in an iterator, neither I'm aware of any special syntax to wrap `subject`, e.g. `isolated { subject(...) { } }`. `first` should work fine for most cases.